src/redux/groups/groupsSliceMiddleware.ts

```diff
@@ -61,6 +60,71 @@ const computeGroups = (
   return groups;
 };
 
+export function treeDataForPropertyOwner(uri: Uri, propertyOwners: PropertyOwners) {
+  const propertyOwner = propertyOwners[uri];
+  return {
+    label: propertyOwner?.name || '',
+    value: uri
+  };
+}
```

Comment:
Since this function is used in other places as well I would think it belongs in some kind of utils file. We might want to consider moving all these to a utils file actually. It is a bit hard to see what this middleware does since there are so many helper functions

Reply:
👍 Moving this and `treeDataFromGroups` to treeUtil.ts. That way, at least all functions which are related to creating the `ThreeNodeData` data structure used for rendering the Mantine tree are removed from this file.  

The reason I put them here to begin with is that I preferred to have all code for generating the data in the same file, to make it easier to overview or find out how the resulting data looks like. For that reason, it also makes sense for me to _not_ move the `computeGroups` and `collectExistingTags` functions somewhere else. But as usual, open for discussion